srml/contract/src/lib.rs

```diff
@@ -14,41 +14,64 @@
 // You should have received a copy of the GNU General Public License
 // along with Substrate. If not, see <http://www.gnu.org/licenses/>.
 
-//! Smart-contract module for runtime; Allows deployment and execution of smart-contracts
-//! expressed in WebAssembly.
+//! # Contract Module
+//! 
+//! The contract module provides functionality to deploy and execute smart-contracts expressed in WebAssembly.
+//! To use it in your module, you need to implement the contract [`Trait`].
+//! The supported dispatchable functions are documented as part of the [`Call`] enum.
+//! 
+//! ## Overview
+//! 
+//! This module provides the ability to create smart-contract accounts and send them messages.
+//! A smart-contract is represented by an account with associated code and storage. 
+//! When a smart-contract account receives a message, the associated code gets executed.
```

Comment:
The situation with using the term "message" is really complicated here.

Initially, I was using the noun "message" and the verb "send" because the word "call" is a bit overloaded:

1. Wasm VM defines a concept of a function and have instructions to **call** a function.
2. EVM also has calls, but they mean a different thing, "to call" means to invoke a different contract. OTOH, Solidity also has functions and a user can call them, which are different kinds of calls.

Once @gavofyork told me that it is more preferred to use "call" terminology and I didn't have a strong feeling on this topic. But now I do, because, now it is possible to say the following (rather exagerrated) sentence:

> In order to **call** a contract we **call** a function ext_**call**. If the **call** fails it reverts all performed changes, and ext_**call** returns a non-zero status code.

Reply:
i see your pt. hmm ill discontinue using "message" since its also a loaded term , in favor of `call`